apps/consulting/next.config.js

```diff
@@ -47,16 +47,30 @@ const nextConfig = {
     ];
   },
   async rewrites() {
-    return [
-      {
-        source: '/plausible/js/script.js',
-        destination: 'https://plausible.io/js/script.js',
-      },
-      {
-        source: '/plausible/api/event',
-        destination: 'https://plausible.io/api/event',
-      },
-    ];
+    // The conventions of this return value are described in the Next.js docs:
+    // https://nextjs.org/docs/api-reference/next.config.js/rewrites
+    return {
+      afterFiles: [
+        // These rewrites are checked after pages/public files are checked but
+        // before dynamic routes.
+        {
+          // Proxying Plausible through Vercel:
+          // https://plausible.io/docs/proxy/guides/vercel
+          source: '/plausible/js/script.js',
+          destination: 'https://plausible.io/js/script.js',
+        },
+        {
+          source: '/plausible/api/event',
+          destination: 'https://plausible.io/api/event',
+        },
+      ],
+      fallback: [
```

Comment:
What I expect this to do:

- User goes to http://quansight-consulting.vercel.app/post/this-post-has-not-yet-been-migrated.
- Url in the browser stays the same.
- User sees the `page-unavailable` page currently in the LLC Storyblok (simple page that says we're working on migrating the content).

What actually happens:

- All the same things as above.
- Page attempts to fetch a non-existent JSON file: `this-post-has-not-yet-been-migrated.json`.
- Because there's no JSON at the end point, the code that fetched the JSON throws an error.
- Some Next.js or React.js machinery catches the error and renders an error message on the page.

Reply:
What I don't understand is why Next.js attempts to fetch that JSON file. The documentation for `getStaticPaths` seems to indicate that if you have `fallback: false`—like we do for all post/[slug] dynamic route—then data-fetching shouldn't happen.

Reply:
I wondered if maybe this was something built in to the way that Next.js works or not. So I created a small test site in which I mimicked several things from this Next.js app:

- [Same Next.js version ](https://github.com/gabalafou/nextjs/blob/main/package.json#L10)
- Same dynamic route. Compare Consulting's [pages/post/[slug].tsx](https://github.com/Quansight/Quansight-website/blob/develop/apps/consulting/pages/post/%5Bslug%5D.tsx) to my test site's [pages/[slug].jsx](https://github.com/gabalafou/nextjs/blob/main/pages/%5Bslug%5D.jsx)
- [Same fallback rewrite](https://github.com/gabalafou/nextjs/blob/main/next.config.js#L6)

So what happens? On my test site, it does exactly what I expected the rewrite to do on the Consulting site. If you visit a URL on my test site, such as [/does-not-exist](https://nextjs-pi-one-76.vercel.app/does-not-exist) that does not match one of the paths return by `getStaticPaths`, then the server falls back to page returned from /page-unavailable. The client app does not attempt to fetch a non-existing JSON file that 404s, and it does not throw an error.

Reply:
It appears that the JSON error is coming about because the 404 page is being (i) erroneously served, _at all_, and (ii) erroneously served _as a JSON payload_.

From the browser console:

```
SyntaxError: Unexpected token < in JSON at position 0
```

From the JSON payload:

```
<!DOCTYPE html><html lang="en" class="text-[62.5%]"><head><meta charSet="utf-8"/><meta name="viewport" content="width=device-width, initial-scale=1"/><title>404: This page could not be found</title> ...
```

----

It also appears to be specific to the `/post/:slug` fallback rewrite, as a [similar rewrite](https://github.com/Quansight/Quansight-website/commit/83d91404d465ee149e071ca8117fec31c271d66e) for a plain, static page [works correctly](https://quansight-consulting-published-602d508u5-quansight.vercel.app/test-missing).